api/namespace.go

```diff
@@ -70,8 +70,15 @@ type Namespace struct {
 	Name        string
 	Description string
 	Quota       string
-	CreateIndex uint64
-	ModifyIndex uint64
+	// TODO: how to encode properly?
+	Capabilities *NamespaceCapabilities
+	CreateIndex  uint64
+	ModifyIndex  uint64
+}
+
+// TODO: is this duplication between struct & api expected/wanted?
```

Comment:
Yes... and yes it's terrible and results in _way_ too much code to write for every feature. 😀  Long term we're hoping that the [OpenAPI](https://github.com/hashicorp/nomad-openapi) project we're doing will eventually be useable for internal use as well.